### DISCUSSION TITLE
Properly setup Active Storage

### DISCUSSION THREAD

**Discussion by ACPK** [thoughtbot/suspenders#1301] [Ideas]
Line #11 in config/environments/production reads:

      config.active_storage.service = :local

With the Heroku setup, shouldn't there be an ENV variable here so users can set it to a third-party service (ex: AWS)?

**Comment by delphaber** [comment]
You are right. `:local` would not work on Heroku.

I think it would be a good idea to use an ENV variable even if the app is not set up on heroku.

```
# in .env
ACTIVE_STORAGE_SERVICE='local'
```

```
# in config/environments/production.rb
config.active_storage.service = ENV.fetch('ACTIVE_STORAGE_SERVICE', :local).to_sym
```

**Comment by stevepolitodesign** [comment]
The [latest release](https://github.com/thoughtbot/suspenders/releases/tag/v20240516.0) made an effort to rely on as much as `rails new` as possible, and to push any infrastructure decisions to the consumer, since we no longer default to hosting on Heroku.

That being said, I don't think there would ever be a case where we'd want to store files locally in `production`.
